File: set_1/easy/javascript/d03b9402-989a-4934-9b9f-c8f2bb839f4e.js
// Solution 1:
function findNumberOfTriangles(arr) {
    let arrLength = arr.length;
    let count = 0;
    let resultantArr = [];

    for (let i = 0; i < arrLength; i++) {
        for (let j = i + 1; j < arrLength; j++) {

            for (let k = j + 1; k < arrLength; k++) {

                if (arr[i] + arr[j] > arr[k]) {
                    resultantArr.push([arr[i], arr[j], arr[k]]);
                    count++;
                }
                if (arr[i] + arr[k] > arr[j]) {
                    resultantArr.push([arr[i], arr[k], arr[j]]);
                    count++;
                }
                if (arr[k] + arr[j] > arr[i]) {
                    resultantArr.push([arr[k], arr[j], arr[i]]);
                    count++;
                }
            }
        }
    }
    return ({
        resultantArr,
        count
    });
}

let arr1 = [4, 6, 3, 7];
resp1 = findNumberOfTriangles(arr1);
console.log(`The number of possible traingles are --> ${resp1.count}`);
console.log("The possible valid traingles are -->",resp1.resultantArr);

File: set_1/easy/python/d03b9402-989a-4934-9b9f-c8f2bb839f4e.py
# Solution 1:
def find_number_of_triangles(list1):
    list1_length, count, resultant_list, result = len(list1), 0, [], dict()
    for i in range(list1_length):
        for j in range(i + 1, list1_length):
            for k in range(j + 1, list1_length):
                if (list1[i] + list1[j] > list1[k] and list1[i] + list1[k] > list1[j] and list1[k]+list1[j] > list1[i]):
                    resultant_list.append([list1[i], list1[j], list1[k]])
                    count += 1
    result["resultant_list"] = resultant_list
    result["count"] = count
    return (result)

list1 = [4, 6, 3, 7]
resp1 = find_number_of_triangles(list1)
print(f"The number of possible traingles are --> {resp1['count']}")
print(f"The possible valid traingles are --> {resp1['resultant_list']}")
